slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/InterchangeExportIntegrationTest.java

```diff
@@ -0,0 +1,128 @@
+
+  private final SignedBlocksDao signedBlocks = new SignedBlocksDao();
+  private final SignedAttestationsDao signedAttestations = new SignedAttestationsDao();
+  private final ValidatorsDao validators = new ValidatorsDao();
+
+  private EmbeddedPostgres setup() throws IOException, URISyntaxException {
+    final EmbeddedPostgres slashingDatabase = EmbeddedPostgres.start();
+
+    final String migrationsFile = Path.of("migrations", "postgresql", "V1__initial.sql").toString();
+
+    final Path schemaPath = Paths.get(Resources.getResource(migrationsFile).toURI());
+
+    final Path migrationPath = schemaPath.getParent();
+
+    final Flyway flyway =
+        Flyway.configure()
+            .locations("filesystem:" + migrationPath.toString())
+            .dataSource(slashingDatabase.getPostgresDatabase())
+            .load();
+    flyway.migrate();
+
+    return slashingDatabase;
+  }
+
+  @Test
+  void canCreateDatabaseWithEntries() throws IOException, URISyntaxException {
+    final EmbeddedPostgres db = setup();
+
+    final String databaseUrl =
+        String.format("jdbc:postgresql://localhost:%d/postgres", db.getPort());
+
+    final Jdbi jdbi = DbConnection.createConnection(databaseUrl, "postgres", "postgres");
+
+    jdbi.useTransaction(
+        h -> {
+          validators.registerValidators(h, List.of(Bytes.fromHexString("0x01")));
+          validators.registerValidators(h, List.of(Bytes.fromHexString("0x02")));
+        });
+
+    final int TOTAL_BLOCKS_SIGNED = 6;
+    jdbi.useTransaction(
+        h -> {
+          for (int i = 0; i < TOTAL_BLOCKS_SIGNED; i++) {
+            signedBlocks.insertBlockProposal(
+                h, new SignedBlock((i % 2) + 1, UInt64.valueOf(i), Bytes.fromHexString("0x01")));
+          }
+        });
+
+    final int TOTAL_ATTESTATIONS_SIGNED = 8;
+    jdbi.useTransaction(
+        h -> {
+          for (int i = 0; i < TOTAL_ATTESTATIONS_SIGNED; i++) {
+            signedAttestations.insertAttestation(
+                h,
+                new SignedAttestation(
+                    (i % 2) + 1,
+                    UInt64.valueOf(i),
+                    UInt64.valueOf(i),
+                    Bytes.fromHexString("0x01")));
+          }
+        });
+
+    final OutputStream exportOutput = new ByteArrayOutputStream();
+    final SlashingProtection slashingProtection =
+        SlashingProtectionFactory.createSlashingProtection(databaseUrl, "postgres", "postgres");
+    slashingProtection.export(exportOutput);
+    exportOutput.close();
+
+    final ObjectMapper mapper = new ObjectMapper().registerModule(new InterchangeModule());
+
+    final InterchangeV5Format outputObject =
+        mapper.readValue(exportOutput.toString(), InterchangeV5Format.class);
+
+    final List<SignedArtifacts> signedArtifacts = outputObject.getSignedArtifacts();
+    assertThat(signedArtifacts).hasSize(2);
+    assertThat(signedArtifacts.get(0).getSignedBlocks()).hasSize(TOTAL_BLOCKS_SIGNED / 2);
```

Comment:
why is the size TOTAL_BLOCKS_SIGNED / 2? Are some not being signed?

Reply:
resolved - but the answer is that there were 2 validators, now both validators get the same number of blocks